indra/tools/reading/readers.py

```diff
@@ -0,0 +1,546 @@
+                    # name a bit, if anything.
+                    if fpath.startswith('PMC'):
+                        file_list.append(fpath)
+                    else:
+                        new_fpath = path.join(self.tmp_dir,
+                                              path.basename(fpath))
+                        shutil.copy(fpath, new_fpath)
+                else:
+                    # Otherwise we need to frame the content in xml and put it
+                    # in a new file with the appropriat name.
+                    old_name = path.basename(fpath)
+                    new_fname = '.'.join(old_name.split('.')[:-1] + ['nxml'])
+                    new_fpath = path.join(self.tmp_dir, new_fname)
+                    with open(fpath, 'r') as f_old:
+                        content = f_old.read()
+                    nxml_str = sparser.make_sparser_nxml_from_text(content)
+                    with open(new_fpath, 'w') as f_new:
+                        f_new.write(nxml_str)
+                    file_list.append(new_fpath)
+            elif all([hasattr(item, a) for a in ['format', 'content', 'id']]):
+                # This implies that it is a text content object, or something
+                # with a matching API.
+                if item.format == formats.XML:
+                    add_nxml_file(
+                        item.id,
+                        zlib.decompress(item.content, 16+zlib.MAX_WBITS)
+                        )
+                elif item.format == formats.TEXT:
+                    txt_bts = zlib.decompress(item.content, 16+zlib.MAX_WBITS)
+                    nxml_str = sparser.make_sparser_nxml_from_text(
+                        txt_bts.decode('utf8')
+                        )
+                    add_nxml_file(item.id, nxml_str.encode('utf8'))
+                elif item.format == formats.JSON:
+                    raise SparserError("I don't know how to handle JSON.")
+                else:
+                    raise SparserError("Unrecognized format %s." % item.format)
+            else:
+                raise SparserError("Unknown type of item for reading %s." %
+                                   type(item))
+        return file_list
+
+    def get_output(self, output_files):
+        "Get the output files as an id indexed dict."
+        reading_data_list = []
+        patt = re.compile(r'(.*?)-semantics.*?')
+        for outpath in output_files:
+            re_out = patt.match(path.basename(outpath))
+            if re_out is None:
+                raise SparserError("Could not get prefix from output path %s."
+                                   % outpath)
+            prefix = re_out.groups()[0]
+            if prefix.startswith('PMC'):
+                prefix = prefix[3:]
+            if prefix.isdecimal():
+                # In this case we assume the prefix is a tcid.
+                prefix = int(prefix)
+
+            with open(outpath, 'r') as f:
```

Comment:
You probably want to open the file in binary mode and then decode the content after reading.

Reply:
done